packages/core/src/animation/Animator.ts

```diff
@@ -0,0 +1,465 @@
+    if (!animatorController) return;
+    const animLayer = animatorController.layers[layerIndex];
+    const theState = animLayer.stateMachine.findStateByName(stateName);
+    theState.frameTime = theState.clip.length * normalizedTime;
+    animLayer._playingState = theState;
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Set the Animator in playback mode.
+   */
+  play(): void {
+    this.recorderMode = AnimatorRecorderMode.Playback;
+  }
+
+  /**
+   * Stops the animator playback mode.
+   */
+  stop(): void {
+    this.recorderMode = AnimatorRecorderMode.Offline;
+  }
+
+  /**
+   * Evaluates the animator component based on deltaTime.
+   * @param deltaTime - The deltaTime when the animation update.
+   * @private
+   */
+  update(deltaTime: number): void {
+    if (this.recorderMode !== AnimatorRecorderMode.Playback) return;
+    const { animatorController } = this;
+    if (!animatorController) return;
+    const { layers } = animatorController;
+    for (let i = 0; i < layers.length; i++) {
+      const isFirstLayer = i === 0;
+      const animLayer = layers[i];
+      if (!animLayer._playingState) {
+        animLayer._playingState = animLayer.stateMachine.states[0];
+      }
+      const currentState = animLayer._playingState;
+      currentState.frameTime += deltaTime / 1000;
+      if (currentState._playType === PlayType.IsFading) {
+        const fadingState = animLayer._fadingState;
+        if (fadingState) {
+          fadingState.frameTime += deltaTime / 1000;
+          if (fadingState.frameTime > fadingState.clipEndTime) {
+            fadingState.frameTime = fadingState.clipEndTime;
+          }
+        }
+      }
+      this._updatePlayingState(currentState, animLayer, isFirstLayer, deltaTime);
+    }
+  }
+
+  /**
+   * crossFade to the AnimationClip by name.
+   */
+  crossFade(
+    name: string,
+    layerIndex: number,
+    normalizedTransitionDuration: number,
+    normalizedTimeOffset: number
+  ): void {
+    const animLayer = this.animatorController.layers[layerIndex];
+    const currentState = animLayer._playingState;
+    if (currentState) {
+      currentState._playType = PlayType.IsFading;
+      const nextState = animLayer.stateMachine.findStateByName(name);
+      if (nextState) {
+        const transition = currentState.addTransition(nextState);
+        this.animatorController.layers[layerIndex]._fadingState = nextState;
+        transition.solo = true;
+        transition.duration = currentState.clip.length * normalizedTransitionDuration;
+        transition.offset = nextState.clip.length * normalizedTimeOffset;
+        transition.exitTime = currentState.frameTime;
+      }
+    }
+  }
+
+  /**
+   * Return the layer by name.
+   * @param name The layer name.
+   */
+  getLayerByName(name: string): AnimatorControllerLayer {
+    return AnimatorControllerLayer.findLayerByName(name);
+  }
+
+  /**
+   * Be called when this instance be enabled.
```

Comment:
internal func write @internal and @override is enough, don't nedd write explantion,unless it is difficult to understand or memo, denpend on your self.

Reply:
done